javaparser-core/src/main/java/com/github/javaparser/utils/PositionUtils.java

```diff
@@ -93,50 +90,50 @@ private static int compare(Node a, Node b, boolean ignoringAnnotations) {
         }
     }
 
-    public static AnnotationExpr getLastAnnotation(Node node) {
+    /**
+     * @return get the last annotation on the node, last meaning "last in reading order."
+     */
+    public static Optional<AnnotationExpr> getLastAnnotation(Node node) {
         if (node instanceof NodeWithAnnotations) {
-            NodeList<AnnotationExpr> annotations = NodeList.nodeList(((NodeWithAnnotations<?>) node).getAnnotations());
-            if (annotations.isEmpty()) {
-                return null;
-            }
+            final NodeList<AnnotationExpr> annotations = nodeList(((NodeWithAnnotations<?>) node).getAnnotations());
             sortByBeginPosition(annotations);
-            return annotations.get(annotations.size() - 1);
-        } else {
-            return null;
+            return annotations.getLast();
         }
+        return Optional.empty();
     }
 
     private static int beginLineWithoutConsideringAnnotation(Node node) {
-        return beginNodeWithoutConsideringAnnotations(node).getRange().get().begin.line;
+        return beginPositionWithoutConsideringAnnotations(node).line;
     }
 
 
     private static int beginColumnWithoutConsideringAnnotation(Node node) {
-        return beginNodeWithoutConsideringAnnotations(node).getRange().get().begin.column;
+        return beginPositionWithoutConsideringAnnotations(node).column;
     }
 
-    private static Node beginNodeWithoutConsideringAnnotations(Node node) {
-        if (node instanceof MethodDeclaration || node instanceof FieldDeclaration) {
-            NodeWithType<?, Type> casted = (NodeWithType<?, Type>) node;
-            return casted.getType();
-        } else if (node instanceof ClassOrInterfaceDeclaration) {
-            ClassOrInterfaceDeclaration casted = (ClassOrInterfaceDeclaration) node;
-            return casted.getName();
-        } else {
-            return node;
-        }
+    private static Position beginPositionWithoutConsideringAnnotations(Node node) {
+        return getLastAnnotation(node)
+                .flatMap(annotation -> annotation.getRange()
+                        .map(range -> range.end))
+                .orElse(node.getRange().get().begin);
```

Comment:
I found this weird method that says "I'll tell you where to find the end of the annotations" but does "here's a node that I consider after the annotations for some types".

I thought I'd better just find the position after the annotations (or, @ftomassetti , should we be after the modifiers too, like `public` `abstract` etc.? In case of a class it even skips `class`!) but that breaks a test:

```java
class Issue40{
    @GET
    @Path("original")
    /**
    * Return the original user.
    */
    public User getOriginalUser(String userName) {
        return userService.getOriginalUser(userName);
    }
}
```
No longer has a comment on the method, it has a comment on the `User` return type. Which is logical since my change moved the "begin position without considering annotations" from the start of `User` to the end of `@Path("original")`.

@ftomassetti - considering that this Javadoc is illegal (it must be before the annotations), where should the comment go?

Reply:
Good question, with no absolute answer, as it frequently the case with comments. I would still feel the comment should go on the method but I am not sure we want to make adapt  comments attributions to treat annotations specially

Reply:
Comment attribution has a setting for this already - I'm struggling with it here :-) Yeah, I think it should go on the method too, which means the whole setting is badly named. It's more like a "put comments on logical places" instead of "put comments on the first following thing."

Reply:
Well, the fact is that users perceive the method to start in a different place w.r.t. where the AST perceives it start. We have to reconciliate these views where they differ